### PR TITLE
SWC-6548, PORTALS-2814 - Add JSONArrayEditor

### PR DESCRIPTION
**Pull request by nickgros**
- Support SWC-6548: Add JSONArrayEditor + Modal wrapper to replace EditJSONListModal GWT widget
 - PORTALS-2814: Fix labels on RJSF buttons

https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/17580037/c98cdc6f-4940-4519-b8d3-baa4f99061ab

MUI Modal handles overflow:
image

To duplicate behavior of this screen, which is used in some parts of the Table Column Schema Editor: 
image

### CODE REVIEW

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  onSubmit: (formData: string[]) => void
+}
+
+const arraySchema: JSONSchema7 = {
+  $schema: 'http://json-schema.org/draft-07/schema#',
+  type: 'array',
+  items: {
+    type: 'string',
+  },
+}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx:24]: Figure it's just easiest to use RJSF here -- we get all of the controls you could ever want for an array editor for free.

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+) {
+  const { value, onChange, onSubmit } = props
+  const [showPasteNewValuesForm, setShowPasteNewValuesForm] = useState(false)
+  const [pastedValues, setPastedValues] = useState('')
+
+  const addPastedValuesToArray = useCallback(() => {
+    if (pastedValues) {
+      parse<string[]>(pastedValues, {
+        complete: result => {
+          console.log(result)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx:38]:
```suggestion
```

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        validator={validator}
+        formData={value}
+        onChange={({ formData }) => onChange(formData)}
+        onSubmit={({ formData }) => onSubmit(formData)}
+        templates={{
+          ArrayFieldDescriptionTemplate: ArrayFieldDescriptionTemplate,
+          ArrayFieldItemTemplate: ArrayFieldItemTemplate,
+          ArrayFieldTemplate: ArrayFieldTemplate,
+          ArrayFieldTitleTemplate: ArrayFieldTitleTemplate,
+          ButtonTemplates: ButtonTemplate,
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx:81]: More work could be done to separate Annotations-specific functionality across all of our RJSF custom components, but this set seems to work for this purpose.

File: packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/template/ArrayFieldItemTemplate.tsx [hunk 4]
```diff
@@ -92,7 +92,7 @@ export default function ArrayFieldItemTemplate<
             disabled={disabled || readonly}
             uiSchema={uiSchema}
             registry={registry}
-            aria-label={`Add New`}
+            aria-label={`Add Item`}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/SchemaDrivenAnnotationEditor/template/ArrayFieldItemTemplate.tsx:95]: The title text is "Add Item", so let's have the label match the title text

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      })
+    } else {
+      setPastedValues('')
+      setShowPasteNewValuesForm(false)
+    }
+  }, [onChange, pastedValues, value])
+
+  return (
+    <Box
+      className="AnnotationEditor"
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx:57]: Update className for this more generic JSONArrayEditor?

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+  const [showPasteNewValuesForm, setShowPasteNewValuesForm] = useState(false)
+  const [pastedValues, setPastedValues] = useState('')
+
+  const addPastedValuesToArray = useCallback(() => {
+    if (pastedValues) {
+      parse<string[]>(pastedValues, {
+        complete: result => {
+          console.log(result)
+          if (result.errors.length > 0) {
+            console.error('Error parsing pasted values', result.errors)
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx:42]: Rather show the error in an alert, and do not clear the user pasted values (and keep the form open) so user can fix the problem and retry

**Review by jay-hodgson — APPROVED** [review]
Alert should be added before merging

File: packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx (hunk 6 above)
Comment (nickgros) [reply packages/synapse-react-client/src/components/JSONArrayEditor/JSONArrayEditor.tsx]: image

**Comment by nickgros** [comment]
After code review changes:

image

**Pull request merged**